pkg/operator/controllers/dnsmasq/machineconfigpool_controller_test.go

```diff
@@ -0,0 +1,166 @@
+	}
+	fakeDh := func(controller *gomock.Controller) *mock_dynamichelper.MockInterface {
+		return mock_dynamichelper.NewMockInterface(controller)
+	}
+	cluster := func(enabled bool) *arov1alpha1.Cluster {
+		return &arov1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+			Status:     arov1alpha1.ClusterStatus{},
+			Spec: arov1alpha1.ClusterSpec{
+				OperatorFlags: arov1alpha1.OperatorFlags{
+					controllerEnabled: strconv.FormatBool(enabled),
+				},
+			},
+		}
+	}
+
+	t.Run("when no cluster resource is present, returns error", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		arocli := fakeAro()
+		mcocli := fakeMco()
+		dh := fakeDh(controller)
+
+		r := &MachineConfigPoolReconciler{
+			log:    logrus.NewEntry(logrus.StandardLogger()),
+			arocli: arocli,
+			mcocli: mcocli,
+			dh:     dh,
+		}
+
+		request := ctrl.Request{}
+
+		_, err := r.Reconcile(context.Background(), request)
+
+		if !kerrors.IsNotFound(err) {
+			t.Errorf("wanted error: cluster not found, got error: %v", err)
+		}
+	})
+
+	t.Run("when controller is disabled, returns with no error", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		arocli := fakeAro(cluster(false))
+		mcocli := fakeMco()
+
+		dh := fakeDh(controller)
+
+		r := &MachineConfigPoolReconciler{
+			log:    logrus.NewEntry(logrus.StandardLogger()),
+			arocli: arocli,
+			mcocli: mcocli,
+			dh:     dh,
+		}
+
+		request := ctrl.Request{}
+
+		_, err := r.Reconcile(context.Background(), request)
+
+		if err != nil {
+			t.Errorf("wanted no error, got error: %v", err)
+		}
+	})
+
+	t.Run("when no MachineConfigPool for request is present, does nothing", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		arocli := fakeAro(cluster(true))
+		mcocli := fakeMco()
+
+		dh := fakeDh(controller)
+
+		r := &MachineConfigPoolReconciler{
+			log:    logrus.NewEntry(logrus.StandardLogger()),
+			arocli: arocli,
+			mcocli: mcocli,
+			dh:     dh,
+		}
+
+		request := ctrl.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: "",
+				Name:      "custom",
+			},
+		}
+
+		_, err := r.Reconcile(context.Background(), request)
+
+		if err != nil {
+			t.Errorf("wanted no error, got error: %v", err)
+		}
+	})
+
+	t.Run("when MachineConfigPool for request exists, reconciles ARO DNS MachineConfig", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		arocli := fakeAro(cluster(true))
+		mcocli := fakeMco(
+			&mcv1.MachineConfigPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "custom",
+					Finalizers: []string{MachineConfigPoolControllerName},
+				},
+				Status: mcv1.MachineConfigPoolStatus{},
+				Spec:   mcv1.MachineConfigPoolSpec{},
+			},
+		)
+
+		dh := fakeDh(controller)
+		dh.EXPECT().Ensure(gomock.Any(), gomock.AssignableToTypeOf(&mcv1.MachineConfig{})).Times(1)
```

Comment:
More of the same - I'd rather see a lookup/verification rather than relying on `Ensure`.